### DISCUSSION TITLE
Connecting a Blob Storage Account

### DISCUSSION THREAD

**Discussion by Jacovlogg** [microsoft/aspire#6361] [Q&A] [answered]
I want to connect my Blob storage account to my apps through Aspire.

I have read the docs and have implemented:
var blobs = builder.AddAzureStorage("storage")
    .AddBlobs("blobs");
    
 In my user secrets I have a connectionstrings:
 "blobs": "<My storage account key>"
 
 This all fails and gives the error:
 _Missing Subscription Configuration_
 
 I have Implemented:
 var blobs = builder.AddAzureStorage("storage")
    .RunAsEmulator()
    .AddBlobs("blobs");
    
But I don't want to emulate locally I want to connect to the live storage while deving locally. Or maybe I am misunderstanding the Emulator?

**So I am not sure what I am missing and what connection string blobs wants and storage wants if not the storage Key?**

**Comment by davidfowl** [comment, marked as answer]
> But I don't want to emulate locally I want to connect to the live storage while deving locally. Or maybe I am misunderstanding the Emulator?

Yes the emulator (https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio%2Cblob-storage) runs locally.

If you want aspire to provision storage the resource in Azure then you can use local provisioning https://learn.microsoft.com/en-us/dotnet/aspire/deployment/azure/local-provisioning.

If you want to use an existing storage resource, then you can use `builder.AddConnectionString` to just set a value of a connection string in the apphost's configuration.

**Comment by Jacovlogg** [comment]
Thanks David!
